distributed/protocol/pandas.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import print_function, division, absolute_import
+
+import pickle
+import sys
+
+import pandas as pd
+
+from .serialize import register_serialization, serialize, deserialize
+
+
+def serialize_pandas_dataframe(df):
+    head = pickle.dumps(df.head(0))
+    headers = []
+    framess = []
+    compression = [None]
+    lengths = [len(head)]
+    for column in df.columns:
+        x = df[column].values
```

Comment:
using ``.values`` will cause coercion to numpy datatypes for most things (except categoricals), IOW, datetimes w/tz are converted to UTC (and thus this is not idempotent). Better to use ``df[column]._values`` which preserves the structure.